server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -75,12 +80,16 @@ public static void initializeFeatureFlags(Settings openSearchSettings) {
      * This method returns true if the value is "true" (case-insensitive),
      * and false otherwise.
      */
-    public static boolean isEnabled(String featureFlagName) {
+    public static boolean isEnabled(String featureFlagName, boolean defaultValue) {
```

Comment:
Hm ... `DATETIME_FORMATTER_CACHING_SETTING` has default value already, I suspect `settings` are `null`, I suspect you need something like that:

```
    public static boolean isEnabled(Setting<Boolean> featureFlag) {
        if (settings != null) {
            return isEnabled(featureFlag.getKey()));
        } else {
            return featureFlag.getDefault(Settings.EMPTY);
        }
    }
```

Reply:
@reta settings are not null, the featurflag class only maintains the setting defined in `opensearch.yml` and not the overall cluster settings initialized by setting module at bootstrap. Therefore, at runtime the setting of the feature flag will be present only if its either a system property or defined in yml file. I couldn't see an alternative way to do it.
If the setting would've been null then the check would've failed in the first condition here
`return settings != null && settings.getAsBoolean(featureFlagName, defaultValue);` but its going in the second condition and couldn't find the specified setting since its not specified in yaml

Reply:
Thanks @CaptainDredge , it seems like setting is completely useless in this case - which is an issue, having an override for `Setting<Boolean>`  and not a `String` would let us to use the default value of the setting

```
public static boolean isEnabled(Setting<Boolean> featureFlag) {
       if ("true".equalsIgnoreCase(System.getProperty(featureFlag.getKey()))) {
            // TODO: Remove the if condition once FeatureFlags are only supported via opensearch.yml
            return true;
        } else if (settings != null) {
           return featureFlag.get(settings);
        } else {
            return featureFlag.getDefault(Settings.EMPTY);
        }
    }
```

Reply:
That's a nice way to do it! Thanks